### PR TITLE
test: add 4-bit ints and addition to type-level programming with Bools test file

### PR DESCRIPTION
**Pull request by xdesou01**

### CODE REVIEW

**Comment by xdesou01** [comment]
Not sure why this included changes (which I thought I reversed) to the example file. Since I'm about to make changes to it anyway, I guess I'll put them here

File: examples/type-level-boolean-programming [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/// A list of type `a` that uses a Boolean at the type level to trackwhether it has
```
Comment (magnus-madsen) [inline examples/type-level-boolean-programming:1]: Why is this here? :)

Comment (xdesou01) [reply examples/type-level-boolean-programming:1]: That's a really good question and I don't know the answer

File: main/test/flix/Test.Type.Bool.flix [hunk 2]
```diff
@@ -447,26 +616,71 @@ mod Test.Type.Bool {
...
+
+    def _bitwiseNot4(_: I4[a, b, c, d]): I4[not a, not b, not c, not d] = I4.I4
+    def _bitwiseOr4(_: I4[a1, b1, c1, d1], _: I4[a2, b2, c2, d2]):
+      I4[a1 or a2, b1 or b2, c1 or c2, d1 or d2] = I4.I4
+    def _bitwiseAnd4(_: I4[a1, b1, c1, d1], _: I4[a2, b2, c2, d2]):
+      I4[a1 and a2, b1 and b2, c1 and c2, d1 and d2] = I4.I4
+    def inc4(_: I4[a, b, c, d]): I4[a xor (b and c and d), b xor (c and d), c xor d, not d] = I4.I4
+    def add4(_: I4[a1, b1, c1, d1], _: I4[a2, b2, c2, d2]):
+      I4[(a1 xor a2 xor ((b1 and b2) or ((b1 xor b2) and ((c1 and c2) or ((c1 xor c2) and (d1 and d2)))))),
+        (b1 xor b2 xor ((c1 and c2) or ((c1 xor c2) and (d1 and d2)))),
```
Comment (magnus-madsen) [inline main/test/flix/Test.Type.Bool.flix:683]: If you add one more space of indent to these lines this becomes easier to read :)

**Review by magnus-madsen — APPROVED** [review]
